mage_ai/data_cleaner/column_type_detector.py

```diff
@@ -66,17 +66,22 @@ def str_in_set(string, string_set):
 def find_syntax_errors(series, column_type):
     if len(series) == 0:
         return pd.Series([])
+    str_series = series.astype(str)
```

Comment:
just to avoid unnecessary regex matching, if the series.dtype is a number type or datetime, we don't need to do regex matching with the str.